crates/circuit/src/circuit_instruction.rs

```diff
@@ -379,16 +380,17 @@ impl CircuitInstruction {
                 let eq = match left {
                     Param::Float(left) => match right {
                         Param::Float(right) => left == right,
-                        Param::ParameterExpression(right) | Param::Obj(right) => {
-                            right.bind(py).eq(left)?
+                        Param::ParameterExpression(right) => {
+                            right == &ParameterExpression::from_f64(*left)
```

Comment:
Not that I think it matters here, but I would have expected:

```suggestion
                            &ParameterExpression::from_f64(*left) == right
```